### DISCUSSION TITLE
Cross-process tracing

### DISCUSSION THREAD

**Discussion by main--** [tokio-rs/tracing#1967] [Q&A] [answered]
I'm wondering if it's possible to sort of "continue" a trace from within another process. So if I have a client that begins a trace and then this client makes a call to a sever, both can report their tracing data (through opentelemetry in my case) and I end up with a trace that includes both the server-side and the client-side action.

I tried sending the Span's Id and specifying it as the parent span on the other side, but `tracing-subscriber`'s `Registry` clearly doesn't like that: https://github.com/tokio-rs/tracing/blob/2b3aa06de43c8e199378b9e4962cb78e658631e5/tracing-subscriber/src/registry/sharded.rs#L273

And in opentelemetry I would also have to make them share a `trace_id` somehow. However, after a quick peek at `tracing-opentelemetry`'s code it seems to me like trying to build a top-level span with `opentelemetry` directly also won't work, since it appears to rely exclusively on the `Registry` to find parent spans.

**Comment by davidbarsky** [comment, marked as answer]
> I'm wondering if it's possible to sort of "continue" a trace from within another process. So if I have a client that begins a trace and then this client makes a call to a sever, both can report their tracing data (through opentelemetry in my case) and I end up with a trace that includes both the server-side and the client-side action.

Sorry for the delay in responding. This is possible, yes.

> I tried sending the Span's Id and specifying it as the parent span on the other side, but tracing-subscriber's Registry clearly doesn't like that:

Span IDs are opaque identifiers that can only be used in a single process—the [Registry's documentation](https://docs.rs/tracing-subscriber/0.3.9/tracing_subscriber/registry/struct.Registry.html#span-id-generation) calls this out, but perhaps this should be in a more prominent location.

> And in opentelemetry I would also have to make them share a trace_id somehow. However, after a quick peek at tracing-opentelemetry's code it seems to me like trying to build a top-level span with opentelemetry directly also won't work, since it appears to rely exclusively on the Registry to find parent spans.

You _will_ be able to use tracing-opentelemetry to propagate trace_id information, primarily through [`OpenTelemetrySpanExt`](https://docs.rs/tracing-opentelemetry/0.17.2/tracing_opentelemetry/trait.OpenTelemetrySpanExt.html) trait. While [this example](https://github.com/tokio-rs/tracing/blob/v0.1.x/examples/examples/opentelemetry-remote-context.rs) does not include Jaeger, it might set you on the right direction.

Does that help a bit?

**Reply by main--** [reply]
Thanks, I think [OpenTelemetrySpanExt](https://docs.rs/tracing-opentelemetry/0.17.2/tracing_opentelemetry/trait.OpenTelemetrySpanExt.html) is exactly what I was missing!
